Source/SwiftLintFramework/Rules/Style/SuperfluousElseRule.swift

```diff
@@ -0,0 +1,152 @@
+import SwiftSyntax
+
+struct SuperfluousElseRule: SwiftSyntaxRule, ConfigurationProviderRule, OptInRule {
+    var configuration = SeverityConfiguration(.warning)
+
+    init() {}
+
+    static var description = RuleDescription(
+        identifier: "superfluous_else",
+        name: "Superfluous Else",
+        description: "Else branches should be avoided when the previous if-block exits the current scope",
+        kind: .style,
+        nonTriggeringExamples: [
+            Example("""
+                if i > 0 {
+                    // comment
+                } else if i < 12 {
+                    return 2
+                } else {
+                    return 3
+                }
+            """),
+            Example("""
+                if i > 0 {
+                    let a = 1
+                    if a > 1 {
+                        // comment
+                    } else {
+                        return 1
+                    }
+                    // comment
+                } else {
+                    return 3
+                }
+            """),
+            Example("""
+                if i > 0 {
+                    if a > 1 {
+                        return 1
+                    }
+                } else {
+                    return 3
+                }
+            """),
+            Example("""
+                if i > 0 {
+                    if a > 1 {
+                        if a > 1 {
+                            // comment
+                        } else {
+                            return 1
+                        }
+                    }
+                } else {
+                    return 3
+                }
+            """, excludeFromDocumentation: true)
+        ],
+        triggeringExamples: [
+            Example("""
+                ↓if i > 0 {
+                    return 1
+                    // comment
+                } else {
+                    return 2
+                }
+            """),
+            Example("""
+                ↓if i > 0 {
+                    return 1
+                } else ↓if i < 12 {
+                    return 2
+                } else if i > 18 {
+                    return 3
+                }
+            """),
+            Example("""
+                ↓if i > 0 {
+                    ↓if i < 12 {
+                        return 5
+                    } else {
+                        ↓if i > 11 {
+                            return 6
+                        } else {
+                            return 7
+                        }
+                    }
+                } else ↓if i < 12 {
+                    return 2
+                } else ↓if i < 24 {
+                    return 8
+                } else {
+                    return 3
+                }
+            """)
```

Comment:
Is it possible to make this work for `break` and `continue` statements in loops also?

Reply:
Yes, it's planned to support these, too, in a follow-up extension to keep the number of OSS findings small.

Reply:
FYI @1in1: #5425